nifi-nar-bundles/nifi-jolt-record-bundle/nifi-jolt-record-processors/src/main/java/org/apache/nifi/processors/jolt/record/JoltTransformRecord.java

```diff
@@ -141,12 +145,23 @@ public class JoltTransformRecord extends AbstractProcessor {
     static final PropertyDescriptor JOLT_SPEC = new PropertyDescriptor.Builder()
             .name("jolt-record-spec")
             .displayName("Jolt Specification")
-            .description("Jolt Specification for transform of record data. This value is ignored if the Jolt Sort Transformation is selected.")
+            .description("Jolt Specification for transform of record data. Only one of 'Jolt Specification' or 'Path To Jolt Specification' may be used. "
+                    + "This value is ignored if the Jolt Sort Transformation is selected.")
             .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .required(false)
             .build();
 
+    static final PropertyDescriptor JOLT_SPEC_FILE = new PropertyDescriptor.Builder()
+            .name("jolt-spec-file")
+            .displayName("Path To Jolt Specification")
+            .description("Path to location of a JOLT specification file. Only one of 'Jolt Specification' or 'Path To Jolt Specification' may be used. "
+                    + "This value is ignored if the Jolt Sort Transformation is selected.")
+            .required(false)
+            .identifiesExternalResource(ResourceCardinality.SINGLE, ResourceType.FILE)
```

Comment:
@mattyb149 Is it possible to enhance the existing `Jolt Specification` property to support multiple resource types? That way the existing property could be either a text string as it is now, or a file reference.